lib/iris/tests/test_concatenate.py

```diff
@@ -335,113 +336,93 @@ def test_order_difference(self):
         result = concatenate(cubes)
         self.assertEqual(len(result), 2)
 
-    def test_masked_fill_value_difference(self):
-        cubes = []
-        y = (0, 2)
-        cube = _make_cube((0, 2), y, 1, mask=True)
-        cube.fill_value = 10
-        cubes.append(cube)
-        cube = _make_cube((2, 4), y, 1, mask=True)
-        cube.fill_value = 20
-        cubes.append(cube)
-        result = concatenate(cubes)
-        self.assertEqual(len(result), 2)
-
-    def test_masked_fill_value_difference__exception(self):
-        cubes = iris.cube.CubeList()
-        y = (0, 2)
-        cube = _make_cube((0, 2), y, 1, mask=True)
-        cube.fill_value = 10
-        cubes.append(cube)
-        cube = _make_cube((2, 4), y, 1, mask=True)
-        cube.fill_value = 20
-        cubes.append(cube)
-        emsg = 'Fill values differ'
-        with self.assertRaisesRegexp(iris.exceptions.ConcatenateError, emsg):
-            cubes.concatenate_cube()
-
 
 class Test2D(tests.IrisTest):
```

Comment:
You never test the fill value when all inputs that get concatenated are unmasked. You could always add a quick check to the test on [L535](https://github.com/bjlittle/iris/blob/fb0a3a51b308d7966c408b6ec2e498667cff29c9/lib/iris/tests/test_concatenate.py#L535)